### ISSUE TITLE
Project Needs Maintainers

### ISSUE THREAD

**Issue by xloem** [apt-mirror/apt-mirror#139]
I noticed this valuable project is without maintainers.

Is anybody able to step up to this task?  Alternatively I propose to either wikify the project so anybody can maintain, or reach out to PR contributors to step up (or both).

I noticed there are a lot of forks with activity, some of the owners are @geosone, @defnn, @bryse-dev, @Stifler6996, @chutzimir, @flavienbwk .

When PRs are no longer merged, projects tend to scatter into diverging forks with none of the bugfixing work consolidated, and a lot of duplicated effort.  There are enough contributors here to find suitable maintenance effort somewhere.

**Comment by Stifler6996** [comment]
I don't believe Perl is the problem, its fine.

My only concern is all these offerings of forks that then "bundle" or docker or somehow add some useless extras, that does not work for most people using this in their chosen Linux.

It means PRs cannot be merged, it means it cannot then be put into mainstream or upstream or used in traditional pkging mechanisms.

I kept mine as is, only fixing the actual problems. But unfortunately the maintainers "have left the building".

**Comment by geosone** [comment]
in my fork i have tried to add all the pull requests into one tree nothing more but i alredy switched to aptly for the mirroring.

**Comment by flavienbwk** [comment]
@Stifler6996 if you don't believe it, it's fine then ;) 
Good community spirit, good luck...

ps: inform yourself on Docker

**Comment by xloem** [comment]
@bholtsclaw are you available to share maintainership with people like geosone or Stifler6969?

Re: docker, it seems to me that's a great improvement but should be provided as optional and contributed to the same tree.

**Comment by rbh-se** [comment]
? The readme-file was updated 14 months ago with:
"We (the current maintainers) lack the time and energy to maintain apt-mirror:"....

**Comment by Fake4d** [comment]
I would love to see the official repo back again - there are a lot of working versions out there - in my company I implemented the version from @Stifler6996 (https://github.com/Stifler6996/apt-mirror) which includes all the fixes is needed to fork the repos for our users.

Is there a way to get the rights on this project to subscribe a new maintainer or contact the official persons? 

The problem is that the official package for the ubuntu distribution is also built from this repo - https://packages.ubuntu.com/source/focal/apt-mirror. If we could not get this repo back in activity maybe I could ask the MOTU people to use a different repo with a bug report for the package.

**Comment by xloem** [comment]
This guy is the current maintainer: https://twitter.com/imbrandon .  I pinged him on github up this thread, but no reply.  Does anybody have a twitter account to maybe try him on there?

**Comment by Stifler6996** [comment]
I looked at the twitter link, all I can say is I hope all is well with him and his family.
🙏 

I will continue to make minimalist and relevant changes to keep this code operational. (To the best we can in a dynamic environment).

Without the ability to (worst case - present a presumably a new package, "apt-mirror-2") to the ecosystem, help is needed. Would you like to help me? Then, we can revive this together as a drop-in replacement.
I would need someone(s) well versed with distributing across several packaging systems.

I am only doing this to help the community, and in my spare time.

The alternate is to just continue to fix problems and ^C/^V to the script file (not ideal) on the OS.

I am happy to take this on, I would like to see a "proportionate response" to make it worthy, since there are already viable replacement options out there.

**Comment by xloem** [comment]
The twitter link looked normal to me.  But no updates since 2019.
I messaged Brandon on facebook, and they replied quickly, saying they can grant ownership to whomever needs it tonight !

**Comment by Stifler6996** [comment]
Glad to here everyone is okay.

**Comment by xloem** [comment]
> Glad to hear everyone is okay.

So you know, I don't know what you are talking about here at all.

**Comment by SadPencil** [comment]
At least, pull requests #120  and #127 should be merged as soon as possible, to keep this repo alive.
These two pull requests fixed my problem when mirroring Debian 10 and Ubuntu 20.04. It works like a charm.
So, if there are some maintainers available, at least merge these two PR to keep this repo alive.

**Comment by xloem** [comment]
Thanks for checking in on this.  The way I got in contact with Brandon was at https://www.facebook.com/brandon.holtsclaw/ .  He's the only one with access to this repository and says he would like to pass it on.

> Sep 24, 2020, 9:33 AM
> You sent
> 2 weeks ago
> Hi Brandon; we're trying to set up a new maintainer for https://github.con/apt-mirror/apt-mirror .  Are you able to share the repository with others?
> Sorry, https://github.com/apt-mirror/apt-mirror
> Brandon Holtsclaw sent
> 2 weeks ago
> Yes, I think so but not at the moment I am not home , and laptop IS at home , when we reunite tonight....  I will grant access/ownership to whomever needs it.
> You sent
> 2 weeks ago
> Thanks so much.  Stifler6996 volunteered at https://github.com/apt-mirror/apt-mirror/issues/139 .  Maybe others will too.
> Project Needs Maintainers · Issue #139 · apt-mirror/apt-mirror
> Brandon Holtsclaw sent
> 2 weeks ago
> Cool
> 6:53 AM
> You sent
> a few seconds ago
> Hey Brandon, we never saw you in the issue thread; did you have trouble accessing it?

**Comment by Paulpatou** [comment]
Any news about Brandon giving you access to the repo ?

**Comment by xloem** [comment]
No reply from Brandom since my last comment.  Maybe he lost access to his facebook account ??  Could something strange be going on?  @Stifler6996, I saw you say "I looked at the twitter link, all I can say is I hope all is well with him and his family.", and the last tweet I'm seeing from Brandon is this one:

> Brandon Holtsclaw
> @imbrandon
> ·
> Mar 11, 2019
> attending Board Elections! at MRIGlobal https://facebook.com/536862509/posts/10156977081362510/

which doesn't indicate that anything is wrong, to me.  Is that the same tweet you are responding to?

He has recent Facebook posts and I sent him a facebook message this morning:
> Brandon, checking in ... is there anything we or I could do to make it easier to address apt-mirorr maintenance?

**Comment by t-b** [comment]
Is Brandon really the only one who can grant additional people access? I've seen at https://github.com/apt-mirror that also @bdrung and @jbouse are in the apt-mirror group. Maybe one of those has enough power?

**Comment by jbouse** [comment]
What is being asked for? I don't have administrative access to the
repository if that is what you're asking.

**Comment by t-b** [comment]
@jbouse yes that was the question. Thanks for confirming.

**Comment by steamboatid** [comment]
what about creating new repo?
I already merging all PR with a very little knowledge at perl.

I simply merge what I thought would be usefull PRs.
please checkout at my PR #150

**Comment by xloem** [comment]
Is the mirror at https://github.com/Stifler6996/apt-mirror a suitable new repo?  @Stifler6996 what was your reason for inferring that Brandon was going through some tough times?

**Comment by SadPencil** [comment]
Stifler6996/apt-mirror is indeed the most suitable repo for now

**Comment by rbh-se** [comment]
28/8, I sent mail to holtsclawb@gmail.com, no answer yet.

**Comment by steamboatid** [comment]
any update on this matter ?

**Comment by foss4ever** [comment]
Any news?

Would it maybe be better to move forward with one of the forks, if no-one can get hold of the maintainer of the original repo?

**Comment by xloem** [comment]
yes, the state of this issue is basically a request from visitors to do one or more of:
- contact brandon
- pick a fork and contribute to or maintain it
- let us know what you believe the best path forward is

I have trouble using the internet and don't understand why I would get confirmation from brandon without followup, or what stifler6969 was responding to when they looked at the twitter link in september 2020.

**Comment by foss4ever** [comment]
@xloem 
RIght, then I personally vote for option 2 since it is already more than a year ago since you first reached out to Brandon. 

IMO it would be better at this point to consider this repo abandoned and move forward with one of the forks (probably @Stifler6996's).

**Comment by steamboatid** [comment]
meanwhile,
I'll keep adding new feature to my own repo and merging pending PR's LOL

**Comment by Palando** [comment]
@steamboatid 
If you are willing to maintain this project, would you ask one of the maintainer if he adds you? As @xloem wrote some posts before, one is reachable as Facebook: https://www.facebook.com/brandon.holtsclaw/ Or if you do not have an account I could ask him.

**Comment by steamboatid** [comment]
> @steamboatid and @Stifler6996 If you are willing to maintain this project, would you ask one of the maintainer if he adds you? As @xloem wrote some posts before, one is reachable as Facebook: https://www.facebook.com/brandon.holtsclaw/ Or if you do not have an account I could ask him.

Hello @Palando,
It's OK for me being a maintainer. 
But, if @xloem  or you already in contact with @bholtsclaw then it will be better either you contact him.

**Comment by Palando** [comment]
Hello @steamboatid
I asked him. I think and hope that he will add you in the next days.

**Comment by foss4ever** [comment]
This repo seems obviously abandoned at this point, so why don't we just move on to one of the forks?

In terms of which fork, I would again suggest: https://github.com/Stifler6996/apt-mirror

**Comment by jbouse** [comment]
> Hm, I tried to reach all three maintainers. One gave an answer and two did not react. But none seems to add new people. 😞 Since is does not work for Debian 11 anymore (at least for a full mirror), may be the Debian project should use a fork of the project. I don't think it's necessary to re-implement everything.

You tried reaching out but only 1 person has any administrative control over the repository while the others (like myself) only have commit write abilities but no administrative abilities. Details matter... I have been responsive to tickets as my time allows but there are limits to what anyone can do until the repository admin grants further rights.

**Comment by bdrung** [comment]
As @jbouse mentioned, only owners have administrative control to add people. Currently only @bholtsclaw and @highdn are owners of apt-mirror.

**Comment by highdn** [comment]
Greetings!
I'd be happy to add more maintainers for the project (unless @bholtsclaw objects).
Please let @bholtsclaw and I know if you are willing to become a maintainer!
Thank you!
-- Dmitriy

**Comment by highdn** [comment]
Dear @jbouse and @bdrung,
You've been granted the Owner role for the project!
Congratulations and please use it wisely!
Thank you for your work!
- Dmitriy

**Comment by steamboatid** [comment]
congrats to @jbouse & @bdrung

**Comment by jbouse** [comment]
As an observation, there are currently 23 PRs open, and some of them make massive changes. I personally don't have time to test and validate these large PRs to make sure things don't break so I'm less inclined to merge them. Those wishing to help the project could aid in testing the changes proposed to aid in getting them potentially merged faster.

I am working full-time and in the process of having a house built so time is stretched thin for me. I'll look at being able to set up a triage group and can add people who would like to help with reviewing PRs to it when I have some time to do so. If we can establish a group of default reviewers we could work to improve the process.

**Comment by SupaHiro** [comment]
> As an observation, there are currently 23 PRs open, and some of them make massive changes. I personally don't have time to test and validate these large PRs to make sure things don't break so I'm less inclined to merge them. Those wishing to help the project could aid in testing the changes proposed to aid in getting them potentially merged faster.
> 
> I am working full-time and in the process of having a house built so time is stretched thin for me. I'll look at being able to set up a triage group and can add people who would like to help with reviewing PRs to it when I have some time to do so. If we can establish a group of default reviewers we could work to improve the process.

Definitely let me know if you need help on reviewing PRs.

**Comment by steamboatid** [comment]
dear @jbouse & @bdrung 

any update on this ?

**Comment by SupaHiro** [comment]
@steamboatid I have begun reviewing open PRs and should hopefully get through them all in the coming weeks. @jbouse I would need at least write or maintain access in order to begin associating the approved PRs with corresponding issues. In the meantime I will continue to provided feedback on any PRs that request a review from apt-mirror/reviewers. Thanks for your patience!

**Comment by jbouse** [comment]
I just made it back to Vancouver this morning after being at sea for a
week. I won't have a computer until I get home Monday evening, so I'll have
a look at permissions when I'm finally home.

**Comment by jbouse** [comment]
@SupaHiro, I've updated @apt-mirror/reviewers to `write` access which should grant the necessary access to handle associating issues and PRs as requested.

Again, if people are looking to help and want to aid in reviewing PRs feel free to step up. There is a backlog, and it will significantly help get through it. I would also like to see some ways to have automated testing that could be done on commit that would also aid in speeding up acceptance of PRs while ensuring we don't have any regressions.

**Comment by nE0sIghT** [comment]
Just in case any interested I started `apt-mirror2` project [1] which is `apt-mirror` reimplementation using Python and suitable as drop-in `apt-mirror` replacement.  
It's pure Python3/asyncio typed OOP implementation.  

I used `apt-mirror2` name for this project, but will immediatelly rename it if any of current authors/maintainers will be against this.

[1] https://gitlab.com/apt-mirror2/apt-mirror2

**Comment by bdrung** [comment]
@nE0sIghT Perfect. Rewriting the tool in a better language like Python, Go or Rust has been on my wishlist for years. apt-mirror2 is a good name. In case of a feature parity dropping this Perl implementation and renaming your rewrite to apt-mirror would be fine IMO.

**Comment by SupaHiro** [comment]
@bdrung  && @nE0sIghT , I'm definitely open to migrating to a python a implementation. Ideally I would prefer to cut at least one more release after reviewing #184 and #185. @rockyluke maybe we could partner on getting this release out?

@chutzimir after we create a final perl release would you be interested in working with me to ensure apt-mirror2 parity?

**Comment by nE0sIghT** [comment]
> to ensure apt-mirror2 parity

I believe that all `apt-mirror` features (at least in main branch) are implemented in `apt-mirror2` now except [http] proxy support for FTP which is not implemented in aioftp [1].  Some new features were added also (like `mirror_path` option).
For now I'm focused on stability and quality (just added initial tests).

[1] https://github.com/aio-libs/aioftp/issues/174

**Comment by clarkema** [comment]
I just came across this thread (ironically in the process of fixing some issues with my local apt-mirror usage)

I use apt-mirror as my primary method of package access for Debian and would be sad to see the demise of the current version. I'd be happy to get involved with maintenance or take over the Perl version in parallel with apt-mirror2  as required.

**Comment by rockyluke** [comment]
> @bdrung && @nE0sIghT , I'm definitely open to migrating to a python a implementation. Ideally I would prefer to cut at least one more release after reviewing #184 and #185. @rockyluke maybe we could partner on getting this release out?

Ready to help!

**Comment by SupaHiro** [comment]
@jbouse When you get a chance would you be able to add @rockyluke to @apt-mirror/reviewers ?

**Comment by SupaHiro** [comment]
> I just came across this thread (ironically in the process of fixing some issues with my local apt-mirror usage)
> 
> I use apt-mirror as my primary method of package access for Debian and would be sad to see the demise of the current version. I'd be happy to get involved with maintenance or take over the Perl version in parallel with apt-mirror2 as required.

Hey @clarkema hoping to understand your use case a bit more. Is your concern that moving to a debian package that would require python3 instead of perl would be incompatible with your current system?

**Comment by nE0sIghT** [comment]
FYI I filled ITP [1] and RFS [2] issues to include apt-mirror2 [3] in the Debian.

[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1080273
[2] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1080305
[3] https://gitlab.com/apt-mirror2/apt-mirror2

**Comment by SupaHiro** [comment]
@nE0sIghT  apologies for the delay I just reached out to @jbouse to see if we can get the additional maintainers added.

**Comment by xloem** [comment]
To clarify, @clarkema said they had time and interest in helping maintain the Perl version here. It is also great there is a Python port elsewhere.

**Comment by SupaHiro** [comment]
Thank you @xloem , I've reached out to @clarkema to confirm interest. If so I will work with @jbouse to get @rockyluke and @clarkema added as additional maintainers!

**Comment by clarkema** [comment]
@SupaHiro I haven't seen any contact other than the above mention, so apologies if I'm missed something. Still happy to be involved though.

**Comment by jbouse** [comment]
> @SupaHiro I haven't seen any contact other than the above mention, so apologies if I'm missed something. Still happy to be involved though.

That is because I'm extremely busy with a full-time job, starting two personal businesses, serving on the board of directors for one non-profit, serving on a planning committee for another non-profit, and having had my development PC crash. I have only had a little spare time. So you haven't missed anything; I've just been busy.

**Comment by clarkema** [comment]
@jbouse I was referring to @SupaHiro's comment that they'd reached out to me.

**Comment by SupaHiro** [comment]
@clarkema There's probably someone on linkedin wondering who this crazy person is :P No worries. I've added you to the project. Welcome!

**Comment by Anduin2017** [comment]
@Stifler6996 seems deleted his account?

**Comment by tfmark** [comment]
> @Stifler6996 seems deleted his account?

Did anyone fork his fork before it was deleted? Is there a good alternative?

**Comment by SadPencil** [comment]
> > @Stifler6996 seems deleted his account?
> 
> 
> 
> Did anyone fork his fork before it was deleted? Is there a good alternative?

This fork contains the latest commint of Stifler996's fork.

https://github.com/SDU-Mirrors/apt-mirror/commit/1ebf610c30c17a10abc6b5c4501ded5ad433253b

**Comment by Anduin2017** [comment]
I found one build from my back up server. Archive this in my gists.

https://gist.aiursoft.cn/anduin/57a393ed8ed343c9bc8a9ae7a30f61e5

Just for archive and nobody is currently maintaining it. However it has the fix from @Stifler6996.

**Comment by nE0sIghT** [comment]
Hello!

I want to inform that apt-mirror2 is on the way to the Debian - all dependencies are accepted to unstable/testing and apt-mirror2 is ready for upload [[1]](https://mentors.debian.net/package/apt-mirror2/).

Because apt-mirror2 developed as drop-in replacement for the apt-mirror there are file conflicts between packages which can be handled either via Conflicts/Replaces or via patching apt-mirror2 to support different file locations.

I think that apt-mirror and apt-mirror2 should Conflict/Replaces on each others. If there are any Debian Developers that supports package replacement please say it in the [1].

[1] https://mentors.debian.net/package/apt-mirror2/

P.S.: It's in the Sid and Ubuntu 25.10 now.

There is ongoing discussion for the apt-mirror2 in Trixie and apt-mirror replacement in the Debian here: https://bugs.debian.org/1104621

**Comment by turbor** [comment]
> Hello!
> 
> I want to inform that apt-mirror2 is on the way to the Debian - all dependencies are accepted to unstable/testing and apt-mirror2 is ready for upload [[1]](https://mentors.debian.net/package/apt-mirror2/).
> 

https://mentors.debian.net/package/apt-mirror2/

I get a 404 for that address? Is the apt-mirror2 project canceled?

**Comment by Anduin2017** [comment]
https://gist.aiursoft.com/anduin/57a393ed8ed343c9bc8a9ae7a30f61e5

This might be last known good fork I kept. I tested it today and seems to be working. (Nobody is maintaining it. Do NOT use in production)

**Comment by nE0sIghT** [comment]
Is there any project member who can react to a security vulnerability I found in the master branch?

I don't see any way to report this issue privately via GitHub (there is no SECURITY.md).

**Comment by nE0sIghT** [comment]
> Is there any project member who can react to a security vulnerability I found in the master branch?

A CVE id CVE-2026-25829 has been assigned to this issue. If there is a project maintainer with whom I can share the details of this vulnerability, please contact me at ykonotopov@gnome.org.
